bin/lib/installation.py

```diff
@@ -302,6 +313,118 @@ def command_config(config: Union[List[str], str]) -> List[str]:
         return config.split(" ")
     return config
 
+class GitHubInstallable(Installable):
```

Comment:
Absolutely fantastic. I have something similar...but not as fully formed and working... on a local branch. This is much better :) Will await this being merged then isue it!